src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector128.Helpers.cs

```diff
@@ -0,0 +1,34 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.Intrinsics.Arm;
+using System.Runtime.Intrinsics.X86;
+
+namespace System.Runtime.Intrinsics
+{
+    // Contains internal helper methods
+    public static partial class Vector128
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static Vector128<T> ConditionalSelectBitwise<T>(Vector128<T> selector, Vector128<T> ifTrue, Vector128<T> ifFalse)
+            where T : struct
+        {
+            Debug.Assert(Sse.IsSupported || AdvSimd.IsSupported);
```

Comment:
This is specific to x86 intrinsics. I do not think it belongs here.

Reply:
Its a helper that provides support for either x86 (`Sse`) or ARM64 (`AdvSimd`).

Reply:
Ok, I have missed that AdvSimd is ARM64.

The point that I was trying to make was that everything on Vector128 works everywhere today. This does not follow the pattern.

Reply:
I had suggested we add this as it represents functionality that we will likely need on repeat occasion and that equally applies to all platforms/architectures.
Its something we should likely eventually support on `Vector<T>` so users don't always need to use the raw hardware intrinsics and codify what is essentially the same path for all platforms.

Reply:
If this is on Vector128, it should have software fallback instead of PNSE.

Reply:
Because its `internal` usage , i followed the pattern used by `internal static class VectorMath` (which is assume caller checks ISA support)

Reply:
> The point that I was trying to make was that everything on Vector128 works everywhere today. This does not follow the pattern.

Maybe we could define a new internal `Vector128Helpers` class to help avoid this issue? I wouldn't want to add a software fallback here without a further discussion around exposing these publicly and what that would mean (which would make it, functionally, a size specific version of `Vector<T>`).

Reply:
Adding it the existing `VectorMath` sounds good to me.